test/runtests.jl

```diff
@@ -33,24 +34,27 @@ end
 
   # single files in top folder
   for file in filter(istestfile, readdir(@__DIR__))
-    (file == basename(@__FILE__)) && continue
+    (file == basename(@__FILE__)) && continue # exclude this file to avoid infinite recursion
     @eval @safetestset $file begin
       include($file)
     end
   end
 
   # test examples
   examplepath = joinpath(@__DIR__, "..", "examples")
-  for file in filter(endswith(".jl"), readdir(examplepath; join=true))
-    filename = basename(file)
-    @eval begin
-      @safetestset $filename begin
-        $(Expr(
-          :macrocall,
-          GlobalRef(Suppressor, Symbol("@suppress")),
-          LineNumberNode(@__LINE__, @__FILE__),
-          :(include($file)),
-        ))
+  for (root, _, files) in walkdir(examplepath)
+    contains(root, "setup") && continue
```

Comment:
This line is confusing me, isn't `examplepath` an absolute path? I think you should be doing `walkdir("../examples")` instead to get relative paths.

Reply:
I guess the exact implementation depends on if we want to be able to run `runtests.jl` from any directory, but either way I think something is a bit off.